src/main/java/org/kiwiproject/beta/slf4j/TimestampingLogger.java

```diff
@@ -0,0 +1,147 @@
+@Beta
+public class TimestampingLogger {
+
+    @SuppressWarnings("NonConstantLogger")
+    private final Logger logger;
+    private long previousTimestamp;
+
+    /**
+     * Create a new instance using the given {@link Logger}.
+     */
+    public TimestampingLogger(Logger logger) {
+        this.logger = requireNotNull(logger);
+    }
+
+    /**
+     * Logs a message and elapsed time message at TRACE level.
+     *
+     * @param message the message or message template
+     * @param args    the arguments to the message template, if any
+     * @see #logElapsed(Level, String, Object...)
+     */
+    public void traceLogElapsed(String message, Object... args) {
+        logElapsed(Level.TRACE, message, args);
+    }
+
+    /**
+     * Logs a message and elapsed time message at DEBUG level.
+     *
+     * @param message the message or message template
+     * @param args    the arguments to the message template, if any
+     * @see #logElapsed(Level, String, Object...)
+     */
+    public void debugLogElapsed(String message, Object... args) {
+        logElapsed(Level.DEBUG, message, args);
+    }
+
+    /**
+     * Logs the given message, and then logs the elapsed time since the previous log.
+     * This results in two separate log messages.
+     *
+     * @param level   the level at which to log the message and elapsed time message
+     * @param message the message or message template
+     * @param args    the arguments to the message template, if any
+     */
+    public void logElapsed(Level level, String message, Object... args) {
+        if (KiwiSlf4j.isEnabled(logger, level)) {
+            var now = System.nanoTime();
+            KiwiSlf4j.log(logger, level, message, args);
+            logElapsedSincePreviousTimestamp(logger, level, now, previousTimestamp);
+            previousTimestamp = now;
+        }
+    }
+
+    private static void logElapsedSincePreviousTimestamp(Logger logger,
+                                                         Level level,
+                                                         long now,
+                                                         long previousTimestamp) {
+        if (previousTimestamp > 0) {
+            var diffInNanos = now - previousTimestamp;
+            KiwiSlf4j.log(logger, level,
+                    " -- Time spent (since previous log): {} nanoseconds / {} millis",
+                    diffInNanos, Duration.ofNanos(diffInNanos).toMillis());
+        } else {
+            KiwiSlf4j.log(logger, level,
+                    " -- Time spent (since previous log): N/A (no previous timestamp)");
+        }
+    }
+
+    /**
+     * Logs a message including elapsed time message at TRACE level.
+     *
+     * @param message the message or message template
+     * @param args    the arguments to the message template, if any
+     * @see #logEmbeddingElapsed(Level, String, Object...)
+     */
+    public void traceLogEmbeddingElapsed(String message, Object... args) {
```

Comment:
For this and the other "embedding" methods, I'm thinking maybe this isn't the best name. Maybe instead of "embedding" the names should include the word "append", e.g. `traceLogAppendingElapsed` or `traceLogAndAppendElapsed`.

Reply:
I prefer `traceLogAppendingElapsed`

Reply:
Yes, I think I agree that "appending" is the better word.

Reply:
Renamed all those methods (and tests too)